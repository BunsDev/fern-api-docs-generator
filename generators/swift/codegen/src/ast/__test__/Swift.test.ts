import { AccessLevel, FunctionModifier } from "..";
import Swift from "../../swift";
import { FileGenerator } from "@fern-api/generator-commons";

describe("Swift Language", () => {
    it("makes file header", () => {
        const output = Swift.makeFileHeader({
            header: "This is the header to a file"
        });
        expect(output.toString()).toMatchSnapshot();
    });

    it("makes import", () => {
        const output = Swift.makeImport({
            packageName: "ExamplePackage"
        });
        expect(output.toString()).toMatchSnapshot();
    });

    it("makes function", () => {
        const output = Swift.makeFunc({
            accessLevel: AccessLevel.Fileprivate,
            modifier: FunctionModifier.Static,
            name: "makePerson",
            async: "async",
            throws: "throws",
            params: [
                Swift.makeParam({
                    title: "name",
                    type: "String"
                }),
                Swift.makeParam({
                    title: "age",
                    type: "Int",
                    defaultValue: "33"
                })
            ]
        });
        expect(output.toString()).toMatchSnapshot();
    });

    it("makes struct", () => {

        const output = Swift.makeStruct({
            accessLevel: AccessLevel.Internal,
            name: "Sample",
            inheritance: [
                Swift.makeClass({
                    name: "Codable"
                })
            ]
        });

        expect(output.toString()).toMatchSnapshot();

    });

    it("makes class", () => {
        const output = Swift.makeClass({
            accessLevel: AccessLevel.Fileprivate,
            name: "Sample",
            functions: [
                Swift.makeFunc({
                    name: "doSomething"
                })
            ]
        });
        expect(output.toString()).toMatchSnapshot();
    });

    it("makes file", () => {
        const output = Swift.makeFile({
            fileHeader: Swift.makeFileHeader({
                header: "// Sample.swift"
            }),
            imports: [
                Swift.makeImport({ packageName: "Foundation" }),
                Swift.makeImport({ packageName: "UIKit" })
            ],
            class: Swift.makeClass({
                accessLevel: AccessLevel.Open,
                name: "Sample",
                functions: [
                    Swift.makeFunc({
                        name: "findStuff"
                    }),
                    Swift.makeFunc({
                        accessLevel: AccessLevel.Public,
                        modifier: FunctionModifier.Static,
                        name: "getStuff",
                        async: "async",
                        throws: "throws",
                        returnObject: "String"
                    })
                ]
            })
        });
        expect(output.toString()).toMatchSnapshot();

        // Make a sample file that actually testable
        FileGenerator.generate({
            fileName: "Sample", 
            node: output, 
            extension: "swift", 
            outputDir: "src/ast/__test__/__snapshots__",
        });

    });
});
