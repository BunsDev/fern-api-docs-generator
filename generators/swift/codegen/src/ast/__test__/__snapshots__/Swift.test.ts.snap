// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swift Language makes class 1`] = `
"fileprivate class Sample {

    func doSomething() {
        print("Hey!")
    }

}"
`;

exports[`Swift Language makes file 1`] = `
"// Sample.swift

import Foundation
import UIKit

open class Sample {

    func findStuff() {
        print("Hey!")
    }

    public static func getStuff() async throws -> String {
        print("Hey!")
    }

}"
`;

exports[`Swift Language makes file header 1`] = `"This is the header to a file"`;

exports[`Swift Language makes function 1`] = `
"fileprivate static func makePerson(name: String, age: Int = 33) async throws {
    print("Hey!")
}"
`;

exports[`Swift Language makes import 1`] = `"import ExamplePackage"`;

exports[`Swift Language makes struct 1`] = `
"fileprivate struct Sample {
    print("Hey")
}
"
`;
